Services/OperationService/Controllers/OperationsController.cs

```diff
@@ -5,30 +5,90 @@
 using System.Threading.Tasks;
 using Interfaces;
 using DTO.BrokerRequests;
+using DTO.RestRequests;
+using System;
+using Microsoft.Extensions.Logging;
 
 namespace OperationService.Controllers
 {
     [ApiController]
     [Route("[controller]")]
     public class OperationsController : ControllerBase
     {
+        private readonly ILogger<OperationsController> logger;
+        public OperationsController([FromServices] ILogger<OperationsController> logger)
+        {
+            this.logger = logger;
+        }
+        
         [Route("instruments/get")]
         [HttpGet]
         public async Task<IEnumerable<Instrument>> GetInstruments(
                 [FromServices] ICommand<GetInstrumentsRequest, IEnumerable<Instrument>> command,
                 [FromQuery] BrokerType broker,
                 [FromQuery] string token,
-                [FromQuery] InstrumentType instrument,
-                [FromQuery] int depth)
+                [FromQuery] InstrumentType instrument
+            )
         {
             return await command.Execute(
                 new GetInstrumentsRequest()
                 {
                     Broker = broker,
                     Token = token,
-                    Depth = depth,
                     Type = instrument
                 });
         }
+
+        [Route("trade")]
+        [HttpPost]
+        public async Task <bool> Trade (
+            [FromServices] ICommand<TradeRequest, bool> command,
+            [FromBody] TradeRequest request
+            )
+        {
+            logger.LogInformation("Trade request of user {request.UserId} received");
+            return await command.Execute(request);
+        }
+
+        [Route("instrument/getFromPortfolio")]
+        [HttpGet]
+        public async Task<Instrument> GetInstrumentFromPortfolio(
+            [FromServices] ICommand<GetInstrumentFromPortfolioRequest, Instrument> command,
+            [FromQuery] string userId,
+            [FromQuery] string figi
+            )
+        {
+            logger.LogInformation("Get instrument {figi} from portfolio of user {userId} received");
+            return await command.Execute(
+               new GetInstrumentFromPortfolioRequest()
+               {
+                   UserId = Guid.Parse(userId),
+                   Figi = figi
+               });
+        }
+
+        [Route("userBalance/get")]
+        [HttpGet]
+        public async Task<UserBalance> GetUserBalance(
+            [FromServices] ICommand<GetUserBalanceRequest, UserBalance> command,
+            [FromQuery] Guid userId
+            )
+        {
+            logger.LogInformation("Get user {request.UserId} balance  received");
+            return await command.Execute(new GetUserBalanceRequest() { UserId = userId });
+        }
+
+        [Route("userBalance/update")]
```

Comment:
Kinda strange that anyone can update balance with any values, but for now it's ok

Reply:
What do you mean? Like without passport data and password check?

Reply:
> What do you mean? Like without passport data and password check?

Without any checks that client really sent some money, but as I said, we don't have any functionality to make it in a different way yet, so it's ok, just mentioning